Testsuite

umlaeute opened the pull request:
trying to improve the test-suite
- no longer depends on bash (using POSIX conformant scripting)
- wrap repetitive code into functions
- unify the test for Linux and Darwin
- allow to override the selected (and excluded) tests via environment variables

sletz commented:
Thanks.
Here is a trace on OSX:

```
./testsuccess osx
FAUST Version 2.38.8
Embedded backends: 
   DSP to C
   DSP to C++
   DSP to CSharp
   DSP to DLang
   DSP to Java
   DSP to Julia
   DSP to old C++
   DSP to Rust
   DSP to SOUL
   DSP to WebAssembly (wast/wasm)
Copyright (C) 2002-2021, GRAME - Centre National de Creation Musicale. All rights reserved. 
location /usr/local/bin/faust
------------------------
Run tests for Darwin
------------------------
Test: faust2android faust2au faust2cagtk faust2caqt faust2csound faust2csvplot faust2faustvst faust2ios faust2jack faust2jackconsole faust2jaqt faust2lv2 faust2max6 faust2msp faust2netjackconsole faust2netjackqt faust2plot faust2puredata faust2sndfile faust2soul faust2supercollider faust2unity faust2vcvrack faust2vst faust2vsti faust2w32max6 faust2w32vst faust2w64max6 faust2w64vst
Test '-midi': faust2cagtk faust2caqt faust2ios faust2jack faust2jackconsole faust2jaqt faust2max6 faust2msp faust2netjackconsole faust2netjackqt faust2soul
Test '-nvoices 8': faust2cagtk faust2caqt faust2ios faust2jack faust2jackconsole faust2jaqt faust2juce faust2lv2 faust2max6 faust2msp faust2netjackconsole faust2netjackqt faust2soul faust2vcvrack
Test '-nvoices 8 -midi': faust2cagtk faust2caqt faust2ios faust2jack faust2jackconsole faust2jaqt faust2juce faust2max6 faust2msp faust2netjackconsole faust2netjackqt faust2soul
Test '-nvoices 8 -effect <foo.dsp> -midi': faust2cagtk faust2caqt faust2ios faust2juce faust2max6 faust2msp faust2netjackconsole faust2netjackqt faust2soul
Test '-soundfile': faust2android faust2bela faust2cagtk faust2caqt faust2ios faust2jack faust2jackconsole faust2jaqt faust2juce faust2max6 faust2msp
Test '-soundfile -resample': faust2cagtk faust2caqt faust2jack faust2jaqt
Test '-httpd': faust2cagtk faust2caqt faust2jack faust2jaqt faust2netjackconsole faust2netjackqt
Test '-osc': faust2android faust2cagtk faust2caqt faust2ios faust2jack faust2jaqt faust2max6 faust2msp faust2netjackconsole faust2netjackqt
Test '-osc -httpd': faust2cagtk faust2caqt faust2jack faust2jaqt faust2netjackconsole faust2netjackqt
Test '-preset auto': faust2cagtk faust2caqt faust2jack faust2jaqt
OK: 'faust2android good.dsp' succeeded!
ERROR: 'faust2au good.dsp' failed
OK: 'faust2cagtk good.dsp' succeeded!
OK: 'faust2caqt good.dsp' succeeded!
OK: 'faust2csound good.dsp' succeeded!
OK: 'faust2csvplot good.dsp' succeeded!
OK: 'faust2faustvst good.dsp' succeeded!
OK: 'faust2ios good.dsp' succeeded!
OK: 'faust2jack good.dsp' succeeded!
OK: 'faust2jackconsole good.dsp' succeeded!
OK: 'faust2jaqt good.dsp' succeeded!
OK: 'faust2lv2 good.dsp' succeeded!
OK: 'faust2max6 good.dsp' succeeded!
OK: 'faust2msp good.dsp' succeeded!
OK: 'faust2netjackconsole good.dsp' succeeded!
OK: 'faust2netjackqt good.dsp' succeeded!
OK: 'faust2plot good.dsp' succeeded!
OK: 'faust2puredata good.dsp' succeeded!
OK: 'faust2sndfile good.dsp' succeeded!
OK: 'faust2soul good.dsp' succeeded!
OK: 'faust2supercollider good.dsp' succeeded!
OK: 'faust2unity good.dsp' succeeded!
OK: 'faust2vcvrack good.dsp' succeeded!
OK: 'faust2vst good.dsp' succeeded!
OK: 'faust2vsti good.dsp' succeeded!
OK: 'faust2w32max6 good.dsp' succeeded!
OK: 'faust2w32vst good.dsp' succeeded!
OK: 'faust2w64max6 good.dsp' succeeded!
OK: 'faust2w64vst good.dsp' succeeded!
OK: 'faust2cagtk -midi good.dsp' succeeded!
OK: 'faust2caqt -midi good.dsp' succeeded!
OK: 'faust2ios -midi good.dsp' succeeded!
OK: 'faust2jack -midi good.dsp' succeeded!
OK: 'faust2jackconsole -midi good.dsp' succeeded!
OK: 'faust2jaqt -midi good.dsp' succeeded!
OK: 'faust2max6 -midi good.dsp' succeeded!
OK: 'faust2msp -midi good.dsp' succeeded!
OK: 'faust2netjackconsole -midi good.dsp' succeeded!
OK: 'faust2netjackqt -midi good.dsp' succeeded!
OK: 'faust2soul -midi good.dsp' succeeded!
OK: 'faust2cagtk -nvoices 8 good.dsp' succeeded!
OK: 'faust2caqt -nvoices 8 good.dsp' succeeded!
OK: 'faust2ios -nvoices 8 good.dsp' succeeded!
OK: 'faust2jack -nvoices 8 good.dsp' succeeded!
OK: 'faust2jackconsole -nvoices 8 good.dsp' succeeded!
OK: 'faust2jaqt -nvoices 8 good.dsp' succeeded!
OK: 'faust2juce -nvoices 8 good.dsp' succeeded!
OK: 'faust2lv2 -nvoices 8 good.dsp' succeeded!
OK: 'faust2max6 -nvoices 8 good.dsp' succeeded!
OK: 'faust2msp -nvoices 8 good.dsp' succeeded!
OK: 'faust2netjackconsole -nvoices 8 good.dsp' succeeded!
OK: 'faust2netjackqt -nvoices 8 good.dsp' succeeded!
OK: 'faust2soul -nvoices 8 good.dsp' succeeded!
OK: 'faust2vcvrack -nvoices 8 good.dsp' succeeded!
OK: 'faust2cagtk -nvoices 8 -midi good.dsp' succeeded!
OK: 'faust2caqt -nvoices 8 -midi good.dsp' succeeded!
OK: 'faust2ios -nvoices 8 -midi good.dsp' succeeded!
OK: 'faust2jack -nvoices 8 -midi good.dsp' succeeded!
OK: 'faust2jackconsole -nvoices 8 -midi good.dsp' succeeded!
OK: 'faust2jaqt -nvoices 8 -midi good.dsp' succeeded!
OK: 'faust2juce -nvoices 8 -midi good.dsp' succeeded!
OK: 'faust2max6 -nvoices 8 -midi good.dsp' succeeded!
OK: 'faust2msp -nvoices 8 -midi good.dsp' succeeded!
OK: 'faust2netjackconsole -nvoices 8 -midi good.dsp' succeeded!
OK: 'faust2netjackqt -nvoices 8 -midi good.dsp' succeeded!
OK: 'faust2soul -nvoices 8 -midi good.dsp' succeeded!
OK: 'faust2cagtk -nvoices 8 -effect good.dsp -midi good.dsp' succeeded!
OK: 'faust2caqt -nvoices 8 -effect good.dsp -midi good.dsp' succeeded!
OK: 'faust2ios -nvoices 8 -effect good.dsp -midi good.dsp' succeeded!
OK: 'faust2juce -nvoices 8 -effect good.dsp -midi good.dsp' succeeded!
OK: 'faust2max6 -nvoices 8 -effect good.dsp -midi good.dsp' succeeded!
OK: 'faust2msp -nvoices 8 -effect good.dsp -midi good.dsp' succeeded!
OK: 'faust2netjackconsole -nvoices 8 -effect good.dsp -midi good.dsp' succeeded!
OK: 'faust2netjackqt -nvoices 8 -effect good.dsp -midi good.dsp' succeeded!
OK: 'faust2soul -nvoices 8 -effect good.dsp -midi good.dsp' succeeded!
ERROR: 'faust2android -soundfile -lang ocpp sound.dsp' failed
OK: 'faust2bela -soundfile -lang ocpp sound.dsp' succeeded!
ERROR: 'faust2cagtk -soundfile -lang ocpp sound.dsp' failed
ERROR: 'faust2caqt -soundfile -lang ocpp sound.dsp' failed
OK: 'faust2ios -soundfile -lang ocpp sound.dsp' succeeded!
ERROR: 'faust2jack -soundfile -lang ocpp sound.dsp' failed
OK: 'faust2jackconsole -soundfile -lang ocpp sound.dsp' succeeded!
ERROR: 'faust2jaqt -soundfile -lang ocpp sound.dsp' failed
OK: 'faust2juce -soundfile -lang ocpp sound.dsp' succeeded!
OK: 'faust2max6 -soundfile -lang ocpp sound.dsp' succeeded!
ERROR: 'faust2msp -soundfile -lang ocpp sound.dsp' failed
OK: 'faust2android -soundfile sound.dsp' succeeded!
OK: 'faust2bela -soundfile sound.dsp' succeeded!
OK: 'faust2cagtk -soundfile sound.dsp' succeeded!
OK: 'faust2caqt -soundfile sound.dsp' succeeded!
OK: 'faust2ios -soundfile sound.dsp' succeeded!
OK: 'faust2jack -soundfile sound.dsp' succeeded!
OK: 'faust2jackconsole -soundfile sound.dsp' succeeded!
OK: 'faust2jaqt -soundfile sound.dsp' succeeded!
OK: 'faust2juce -soundfile sound.dsp' succeeded!
OK: 'faust2max6 -soundfile sound.dsp' succeeded!
ERROR: 'faust2msp -soundfile sound.dsp' failed
OK: 'faust2cagtk -soundfile -resample sound.dsp' succeeded!
ERROR: 'faust2caqt -soundfile -resample sound.dsp' failed
OK: 'faust2jack -soundfile -resample sound.dsp' succeeded!
ERROR: 'faust2jaqt -soundfile -resample sound.dsp' failed
OK: 'faust2juce -soundfile -standalone sound.dsp' succeeded!
OK: 'faust2cagtk -httpd good.dsp' succeeded!
OK: 'faust2caqt -httpd good.dsp' succeeded!
OK: 'faust2jack -httpd good.dsp' succeeded!
OK: 'faust2jaqt -httpd good.dsp' succeeded!
OK: 'faust2netjackconsole -httpd good.dsp' succeeded!
OK: 'faust2netjackqt -httpd good.dsp' succeeded!
OK: 'faust2android -osc good.dsp' succeeded!
OK: 'faust2cagtk -osc good.dsp' succeeded!
OK: 'faust2caqt -osc good.dsp' succeeded!
OK: 'faust2ios -osc good.dsp' succeeded!
OK: 'faust2jack -osc good.dsp' succeeded!
OK: 'faust2jaqt -osc good.dsp' succeeded!
OK: 'faust2max6 -osc good.dsp' succeeded!
ERROR: 'faust2msp -osc good.dsp' failed
OK: 'faust2netjackconsole -osc good.dsp' succeeded!
OK: 'faust2netjackqt -osc good.dsp' succeeded!
OK: 'faust2cagtk -osc -httpd good.dsp' succeeded!
OK: 'faust2caqt -osc -httpd good.dsp' succeeded!
OK: 'faust2jack -osc -httpd good.dsp' succeeded!
OK: 'faust2jaqt -osc -httpd good.dsp' succeeded!
OK: 'faust2netjackconsole -osc -httpd good.dsp' succeeded!
OK: 'faust2netjackqt -osc -httpd good.dsp' succeeded!
OK: 'faust2cagtk -preset auto good.dsp' succeeded!
OK: 'faust2caqt -preset auto good.dsp' succeeded!
OK: 'faust2jack -preset auto good.dsp' succeeded!
OK: 'faust2jaqt -preset auto good.dsp' succeeded!
```

so faust2dummy, faust2dummymem, faust2api... related tests are not executed anymore. Can you have  look? Thanks.

umlaeute commented:
thanks for noticing. i'll check this.
is there a trace of an original testrun somewhere available (esp. on macOS which i don't have readily at hand)

note that the argument (`osx`) is a no-op (this is not something i introduced; it wasn't doing anything before i changed things, even though the documentation says otherwise).

umlaeute commented:
umpf. the bug was just a typical negative logic error (tests show be included if they are not in the list of excluded tests rather than not not).  hopefully fixed with 0dfebe9

sletz commented:
Thanks. Testing OK on maxOS now. I guess you have tested on Linux, everything is OK on your side?

sletz commented:
Merged, thanks !
